ui/admin/app/routes/scopes/scope/auth-methods/index.js

```diff
@@ -4,10 +4,109 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
 export default class ScopesScopeAuthMethodsIndexRoute extends Route {
+  // =services
+
+  @service store;
+  @service can;
+  @service router;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    types: {
+      refreshModel: true,
+      replace: true,
+    },
+    primary: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
   // =methods
 
+  /**
+   * Loads queried auth-methods and the number of auth-methods under current scope.
+   * @returns {Promise<{totalItems: number, authMethods: [AuthMethodModel], authMethodsExist: boolean }> }
+   */
+  async model({ search, types, primary, page, pageSize }) {
+    const scope = this.modelFor('scopes.scope');
+    const { id: scope_id } = scope;
+    const filters = {
+      scope_id: [{ equals: scope_id }],
+      type: [],
+      is_primary: [],
+    };
+
+    types.forEach((type) => {
+      filters.type.push({ equals: type });
+    });
+    if (primary.length == 1) {
+      primary.forEach((val) => {
+        filters.is_primary.push({ equals: val });
+      });
+    }
+
+    let authMethods;
+    let totalItems = 0;
+    if (this.can.can('list model', scope, { collection: 'auth-methods' })) {
+      authMethods = await this.store.query('auth-method', {
+        scope_id: 'global',
+        recursive: true,
+        query: { search, filters },
+        page,
+        pageSize,
+      });
+      totalItems = authMethods.meta?.totalItems;
+    }
+    const authMethodsExist = await this.getAuthMethodsExist(
+      scope_id,
+      totalItems,
+    );
```

Comment:
Should this query also be inside our `can list` check? If they cannot list auth-methods, the call will fail.

Reply:
great catch thank you!! this has been updated!